### ISSUE TITLE
Certain images failed to download

### ISSUE THREAD

**Issue by LucionGid** [anekos/Ank-Pixiv-Tool#147]
Hello everyone.

I am a long-term user this wonderful addon from its early release days. Thus, it is great to see the updated webextension of ankpixiv being published officially.

However, i have noticed that there are certain images that can't be downloaded (ankpixiv gives me the "Download Failed" notification whenever i try to download them). The link below is just one example

https://www.pixiv.net/member_illust.php?mode=medium&illust_id=65633752

This has also happened to me in the good old days with ankpixiv. The solution was simply wait sometime until the images become downloadable again. However, it seems that the same trick won't work this time. Also, the number of images giving me the "download failed" notification seems to increase day by day.

I don't know what cause this to happen, and how to fix this. Plus, i have tried some similar extensions with the same functionality (on both FF and Chrome) and they seem to get the images just fine. Any help or suggestion would be greatly appreciated.

Thank you for reading, and have a nice day.

**Comment by toshiya44** [comment]
I didn't get any errors on that image. Are you using any other extensions? What is you browser and OS?

**Comment by LucionGid** [comment]
@toshiya44 : Thank you for your reply. I also use some other extensions beside ankpixiv, but i don't think that is the cause, as the issue only happens to a limited set of images.

I am using Firefox Developer Edition version 57.0b12 (64-bit). The OS is Windows 10 Enterprise Version 1607 build 14393.1770.

**Comment by LucionGid** [comment]
I've figured out why this happens. My filename format was set to included the artists' "memoized-names", and all of the undownloadable images were from artists whose "memoized-name" were number-only (e.g., for the aforementioned link, the artist's memoized name is "59", which is number-only). 

After removing the "memoized-names" from the filename format, i can download the images without problem, so the issue seems to be that ankpixiv somehow cannot extract the number-only "memoized-name". It would be nice if this issue is fixed in future releases.

Cheers.

**Comment by QuessParaya** [comment]
I also get this but the memoized-name solution does not work for me.

I also get these errors in the console:
`16:43:26.302 Error:  (0) : https://i.pximg.net/img-original/img/2017/10/28/17/26/00/65629541_p0.png
Stack trace:
request/</xhr.ontimeout@moz-extension://986d00ab-d458-452a-adb0-77f7ad088792/js/remote.js:60:16
 1 utils.js:401:7
	downloadTargets moz-extension://986d00ab-d458-452a-adb0-77f7ad088792/js/utils.js:401:7
`

`16:43:26.303 dwer Error:  (0) : https://i.pximg.net/img-original/img/2017/10/28/17/26/00/65629541_p0.png
Stack trace:
request/</xhr.ontimeout@moz-extension://986d00ab-d458-452a-adb0-77f7ad088792/js/remote.js:60:16
 1 background.js:514:11
	opDownloadQueue/< moz-extension://986d00ab-d458-452a-adb0-77f7ad088792/js/background.js:514:11
`

**Comment by LucionGid** [comment]
@QuessParaya : May i know your ankpixiv filename's format? I can download that just fine.

**Comment by QuessParaya** [comment]
@LucionGid : AnkPixiv/?site-name?/?member-id?/?illust-year?-?illust-month?-?illust-day? (?illust-id?) ?title?

**Comment by LucionGid** [comment]
I have tried your filename's format and it works without any issue.

I think you should perform a trial-and-error test to see which part of the filename's format is causing the trouble at your end.

**Comment by QuessParaya** [comment]
I just found out that it does not work well with uMatrix addon and NoScript. When I remove NoScript, leaving uMatrix, and then restart the browser, it still does not work. When I finally removed uMatrix, it works even with the memoized-name.

**Comment by LucionGid** [comment]
@QuessParaya : As I mentioned before, the ?memoized-name? option only struggles with artists whose memoized-names are number-only. The ?member-name? option, on the other hand, cannot work with artists whose member-names start with "." (e.g., there is an artist whose member-name is .suke and the ?member-name? option gives me an error in that case). Since the artist's memoized-name in your sample link is pretty standard (including both characters and numbers, and doesn't start with "."), it is no surprise that the memoized-name option work.

As for your case, if you use ankpixiv a lot, i recommend that you have two different versions of firefox, one for normal web browsing (in which your usual addons like uMatrix are installed) and the other for pixiv/twitter/etc images downloading (in which ankpixiv and some non-conflict addons are installed). Just remember to import your ankpixiv download history if the latter is not your original one.

As an example, you can install both the normal firefox and firefox developer edition. They can run in parallel just fine.

**Comment by QuessParaya** [comment]
I found a solution without disabling both!

It seems that uMatrix is the main reason this addon does not work. I was observing the uMatrix logs and apparently, it keeps blocking xhr's being sent by the extension. After digging further, I found out that there is an [open issue in uMatrix where it blocks the moz-extension scheme](https://github.com/violentmonkey/violentmonkey/issues/231). Adding `matrix-off: moz-extension-scheme true` to your ruleset or installing the webextensions version of uMatrix can fix it (I did the former).

I hope this helps others having similar issues.
